src/python/pants/backend/codegen/soap/target_types.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    AllTargets,
+    Dependencies,
+    MultipleSourcesField,
+    SingleSourceField,
+    Target,
+    TargetFilesGenerator,
+    Targets,
+)
+from pants.util.logging import LogLevel
+
+
+class WsdlDependenciesField(Dependencies):
+    pass
+
+
+class AllWsdlTargets(Targets):
+    pass
+
+
+@rule(desc="Find all WSDL sources in a project", level=LogLevel.DEBUG)
+def find_all_wsdl_targets(all_targets: AllTargets) -> AllWsdlTargets:
+    return AllWsdlTargets([tgt for tgt in all_targets if tgt.has_field(WsdlSourceField)])
+
+
+# -----------------------------------------------------------------------------------------------
+# `wsdl_source` target
+# -----------------------------------------------------------------------------------------------
+
+
+class WsdlSourceField(SingleSourceField):
+    expected_file_extensions = (".wsdl",)
+
+
+class WsdlSourceTarget(Target):
+    alias = "wsdl_source"
```

Comment:
Are the semantics of `wsdl` such that it always makes sense to compile file-at-a-time?

Reply:
It does as in a way each WSDL has a `targetNamespace` attribute which could be different or that you may be interested in overriding using the `jvm_package` / `jvm_module` fields.

Also when having a few WSDL files in the same folder and those WSDLs have embedded XSDs with names that may colide. Treating them individually helps removing class name collisions as at the moment can not guarantee those XSDs are defining the same types (could probably be added later but this was a first go at it).